File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        if (intervals.size() == 0)
        {
            return {newInterval};
        }
        vector<vector<int>> answer;
        auto first = find_if(intervals.begin(), intervals.end(), [&](auto& i) { return i[1] >= newInterval[0]; });
        auto last = find_if(intervals.rbegin(), intervals.rend(), [&](auto& i) { return i[0] <= newInterval[1]; });
                
        copy(intervals.begin(), first, back_inserter(answer));
        answer.push_back({min(first == intervals.end() ? INT_MAX : (*first)[0], newInterval[0]), max(last == intervals.rend() ? INT_MIN : (*last)[1], newInterval[1])});
        copy(intervals.begin() + (intervals.size() - distance(intervals.rbegin(), last)), intervals.end(), back_inserter(answer));
        return answer;
    }
};

File: 0057-insert-interval/0057-insert-interval.cs
public class Solution {
    public int[][] Insert(int[][] intervals, int[] newInterval)
    {
        List<int[]> answer = new();
        var i = 0;

        for (; i < intervals.Length && intervals[i][1] < newInterval[0]; ++i)
        {
            answer.Add(intervals[i]);
        }

        for (; i < intervals.Length && intervals[i][0] <= newInterval[1]; ++i)
        {
            newInterval[0] = Math.Min(newInterval[0], intervals[i][0]);
            newInterval[1] = Math.Max(newInterval[1], intervals[i][1]);
        }
        answer.Add(newInterval);

        for (; i < intervals.Length; ++i)
        {
            answer.Add(intervals[i]);
        }

        return answer.ToArray();
    }
}